ui/src/tool-panel/location-track/location-track-location-infobox.tsx

```diff
@@ -169,16 +174,67 @@ export const LocationTrackLocationInfobox: React.FC<LocationTrackLocationInfobox
         setStartingSplitting(true);
         getSplittingInitializationParameters(draftLayoutContext(layoutContext), locationTrack.id)
             .then((splitInitializationParameters) => {
-                if (startAndEndPoints?.start && startAndEndPoints?.end && trackNumber) {
+                if (
+                    startAndEndPoints?.start &&
+                    startAndEndPoints?.end &&
+                    trackNumber &&
+                    extraInfo
+                ) {
+                    const switches = splitInitializationParameters?.switches || [];
+                    const getSwitchName = (switchId: LayoutSwitchId) =>
+                        switches.find((sw) => sw.switchId == switchId)?.name;
+                    const endPointTerm = t('tool-panel.location-track.splitting.endpoint');
+                    const noNameErrorTerm = 'ERROR: no name';
+                    const startSplitPointName =
+                        getSplitPointName(extraInfo.startSplitPoint, getSwitchName, endPointTerm) ||
+                        noNameErrorTerm;
+                    const endSplitPointName =
+                        getSplitPointName(extraInfo.endSplitPoint, getSwitchName, endPointTerm) ||
+                        noNameErrorTerm;
+
+                    const duplicates = splitInitializationParameters?.duplicates || [];
+                    const duplicatesWithNames: SplitDuplicate[] = duplicates.map((duplicate) => {
+                        return {
+                            ...duplicate,
+                            status: {
+                                ...duplicate.status,
+                                startSplitPoint: duplicate.status.startSplitPoint && {
+                                    ...duplicate.status.startSplitPoint,
+                                    name:
+                                        getSplitPointName(
+                                            duplicate.status.startSplitPoint,
+                                            getSwitchName,
+                                            endPointTerm,
+                                        ) || noNameErrorTerm,
+                                },
+                                endSplitPoint: duplicate.status.endSplitPoint && {
+                                    ...duplicate.status.endSplitPoint,
+                                    name:
+                                        getSplitPointName(
+                                            duplicate.status.endSplitPoint,
+                                            getSwitchName,
+                                            endPointTerm,
+                                        ) || noNameErrorTerm,
+                                },
+                            },
+                        };
```

Comment:
Tämä on kyllä jotenkin todella monimutkainen rakenne siihen, että saadaan korvattua splitpointtien nimet 🙈 Mutta enpä kyllä itsekään nopeasti keksi mitään kevyempää. Tätä voisi tosin ehkä pilkkoa funktioiksi jotenkin (esim. vaikka jonkinlaiseksi `SplitPoint`:in ottavaksi `replaceSplitPointName`-funktioksi,) niin tämä ei olisi ihan niin monsteri kaikkine sisennyksineen ja sijoituksineen